test/TestExtensions/API.Test/VSSolutionHelper.cs

```diff
@@ -478,7 +478,7 @@ public static async Task<EnvDTE.Project> GetProjectAsync(EnvDTE80.Solution2 solu
 
             var projectItem = solutionFolderProject.ProjectItems
                 .Cast<EnvDTE.ProjectItem>()
-                .FirstOrDefault(project => project.Name.Equals(projectName, StringComparison.OrdinalIgnoreCase));
+                .FirstOrDefault(project => { ThreadHelper.ThrowIfNotOnUIThread(); return project.Name.Equals(projectName, StringComparison.OrdinalIgnoreCase); });
```

Comment:
I think this check `ThreadHelper.ThrowIfNotOnUIThread();` is redundant here because the caller is on the main thread already unless I am missing something basic.

Reply:
Same as the above. False positive by the analyzer. 

See https://github.com/microsoft/vs-threading/issues/577